File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    
    static String answer = "";
    
    public String solution(String[] participant, String[] completion) {
        
        HashMap<String, Integer> set = new HashMap<>();
        
        for (String part : participant ){
            set.put(part, set.getOrDefault(part, 0) + 1);
        }
        
        for (String comp : completion) {
            set.put(comp, set.get(comp) - 1);
        }
        
        for (String one : set.keySet()){
            if(set.get(one) > 0){
                return one;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(part, comp) {
    
    part.sort();
    comp.sort();
    
    for (let i = 0; i < part.length; i++) {
        if (part[i] != comp[i]) {
            return part[i];
        }
    }
    return comp[comp.length - 1]
}